17.md

```diff
@@ -0,0 +1,148 @@
+NIP-17
+======
+
+Private Direct Messages and Group DMs
+-------------------------------------
+
+`draft` `optional`
+
+This NIP defines an encrypted direct messaging scheme for 2+ participants using [NIP-44](44.md) encryption and [NIP-59](59.md) seals and gift wraps. 
```

Comment:
I have not ready all 249 comments on this PR, but I see group size being discussed a lot. I just wanted to add my vote that we specify an upper limit on group size in this NIP, and create a separate NIP for larger encrypted groups. The upper limit will have to be somewhat arbitrary but it will be something around which apps can standardize their UI. I don't want every app to have a different limit on group size and I don't want every app to have _no_ limit on group size because both of those will create bad experiences for the user.

For large groups the expectation of privacy decreases and developer implementation difficulty increases. A large encrypted group will likely want better performance characteristics at the cost of privacy. This NIP strikes a nice balance between simplicity and privacy for small groups.

```suggestion
This NIP defines an encrypted direct messaging scheme for 2-16 participants using [NIP-44](44.md) encryption and [NIP-59](59.md) seals and gift wraps. 
```

In secure scuttlebutt the original group DM spec only supported 8 participants, and it was fine for like 80% of use cases. I think doubling that to 16 will cover like 90% of real world use cases for group DMs, so that's my arbitrary, anecdotal suggestion here.

Reply:
I don't know if we need to specify it on the spec. It's hard enough to sign, encrypt, download DM relays and publish to each recipient's relays separately that this alone might serve as a better blocker against big groups than anything else.

Reply:
Ok, I made a few changes to demphasize Groups from this NIP.